maintenance/code-backports.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+# silence debug output from VCS
+beQuiet="--quiet"
+#beQuiet=""
+version=`basename "$1" | sed s/squid-//`
+vnext=$(( $version + 1 ))
+srcbranch=v$vnext
+
+cd $1
+
+# Dev branch does not receive backports
+test -f .BASE && exit 0
+
+# clean the current workspace
+gitCleanWorkspace ()
+{
+  git clean $beQuiet -xdf --exclude="\.BASE"
+  git checkout $beQuiet -- .
+}
+
+if ! test -d .git ; then
+  echo "ERROR: missing git repository"
+  exit 1;
+fi
+if ! test -d ../squid-$vnext ; then
+  echo "ERROR: missing git repository to backport from"
+  exit 1;
+fi
+if test -e ../squid-$vnext/.BASE ; then
+  srcbranch=`cat ../squid-$vnext/.BASE`
+fi
+
+echo "FOUDN srcbranch=$srcbranch"
+
+# Prepare backports branch for updates
+git checkout $beQuiet v$version-next-backports 2>/dev/null &&
+  gitCleanWorkspace &&
+    git rebase origin/v$version 2>/dev/null || ( git rebase --abort 2>/dev/null ; exit 1 )
+
+# handler on successful backport
+doneCherryPick ()
+{
+  sha="$1"
+  prnum="$2"
+  version="$3"
+
+  git push $beQuiet >/dev/null
+  # TODO automatically remove label from PRs on successful commit+push to vN-next-backports
```

Comment:
I disagree with this TODO: We should not remove the label just because we managed to push a commit to some unofficial branch. Until the backported commit is _officially merged_, the work prescribed by that label is not done, and, hence, that human-set label should not be automatically removed.

Reply:
dropped